File: wf-beneficiary-management/template/java/model/response/Result.java
package {basePackage}.wf.model.response;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import lombok.Data;

/**
 * Common API result structure for all WF API responses.
 */
@Data
public class Result {

    /**
     * Result status: S (Success), F (Failure), U (Unknown/Retry)
     */
    private String resultStatus;

    /**
     * Result code
     */
    private String resultCode;

    /**
     * Result message
     */
    private String resultMessage;

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: new-skills/references/beneficiary/golang/model/response/result.go
package response

// Result represents the common result structure in WF API responses.
type Result struct {
	// ResultStatus is the result status: S (success), F (failure), U (unknown)
	ResultStatus string `json:"resultStatus"`

	// ResultCode is the result code (结果码)
	ResultCode string `json:"resultCode"`

	// ResultMessage is the result message (结果信息)
	ResultMessage string `json:"resultMessage"`
}

// IsSuccess returns true if the result status is "S" (success).
func (r *Result) IsSuccess() bool {
	return r.ResultStatus == "S"
}

// IsFailure returns true if the result status is "F" (failure).
func (r *Result) IsFailure() bool {
	return r.ResultStatus == "F"
}

// IsUnknown returns true if the result status is "U" (unknown/retryable).
func (r *Result) IsUnknown() bool {
	return r.ResultStatus == "U"
}
